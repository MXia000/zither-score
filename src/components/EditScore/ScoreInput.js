import React, { Component } from "react";
import SingleLineScoreInput from "./SingleLineScoreInput";
import Container from "@material-ui/core/Container";
import ControlPointIcon from "@material-ui/icons/ControlPoint";

class ScoreInput extends Component {
  constructor(props) {
    super(props);
    this.state = {
      line: 0,
      lineLength: []
    };

    this.handleNewLine = this.handleNewLine.bind(this);
  }

  handleNewLine = event => {
    event.preventDefault();
    this.setState(preState => ({
      line: this.state.line + 1,
      lineLength: [...preState.lineLength, this.state.line]
    }));
  };

  render() {
    return (
      <Container maxWidth="lg">
        <button onClick={this.handleNewLine}>
          <ControlPointIcon />
        </button>
        <form>
          <SingleLineScoreInput
            key={this.state.line}
            lineLength={this.state.lineLength}
          />
        </form>
      </Container>
    );
  }
}

export default ScoreInput;
