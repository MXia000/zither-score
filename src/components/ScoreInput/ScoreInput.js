import React, {Component} from 'react';
import { withRouter} from 'react-router-dom';
import { ToggleButtonGroup, ToggleButton} from 'react-bootstrap/';

const inputStyle = {
    fontSize: '12px',
    width: '15px',
    height: '20px',
    position: 'absolute center'
};

const buttonStyle = {
    width:'30px',
    height: '40px',
    padding: '5px',
    margin: '1px',
    position: 'relative'
};

const pStyle = {
    fontSize: '25px',
    float: 'left',
    display: 'inline'
};

const toggleStyle = {
    float: 'left',
    display: 'inline'
};

class ScoreInput extends Component {
    constructor(props){
        super(props);
        this.state = {
            isClick: false
        }

        this.handleToolBar = this.handleToolBar.bind(this);
    }

    handleToolBar() {
        this.setState({
            isClick: !this.state.isClick
        })
    }

    numberOnly(num) {
        var ch = String.fromCharCode(num.which);
        if(!(/[0-7]/.test(ch))) {
            num.preventDefault();
        }
    }
    render () {
        return (
            <div> 
                <div>
                    <p style={pStyle}>|</p>
                    <ToggleButtonGroup type="checkbox" defaultValue={['', '']} style={toggleStyle}>
                        <ToggleButton onClick={this.handleToolBar} value={'0, 0'} style={buttonStyle}>
                            <input style={inputStyle} className="note" type="text" maxLength="1" onKeyPress={(inp) => this.numberOnly(inp)}></input> 
                        </ToggleButton>
                    
                        <ToggleButton value={'0, 1'} style={buttonStyle}>
                            <input style={inputStyle} className="note" type="text" maxLength="1" onKeyPress={(inp) => this.numberOnly(inp)}></input> 
                        </ToggleButton>
                    
                        <ToggleButton value={'0, 2'} style={buttonStyle}>
                            <input style={inputStyle} className="note" type="text" maxLength="1" onKeyPress={(inp) => this.numberOnly(inp)}></input> 
                        </ToggleButton>
                    
                        <ToggleButton value={'0, 3'} style={buttonStyle}>
                            <input style={inputStyle} className="note" type="text" maxLength="1" onKeyPress={(inp) => this.numberOnly(inp)}></input> 
                        </ToggleButton>
                    </ToggleButtonGroup>
                    <p style={pStyle}>|</p>
                    <ToggleButtonGroup type="checkbox" defaultValue={['', '']} style={toggleStyle}>
                        <ToggleButton value={'1, 0'} style={buttonStyle}>
                            <input style={inputStyle} className="note" type="text" maxLength="1" onKeyPress={(inp) => this.numberOnly(inp)}></input> 
                        </ToggleButton>
                    
                        <ToggleButton value={'1, 1'} style={buttonStyle}>
                            <input style={inputStyle} className="note" type="text" maxLength="1" onKeyPress={(inp) => this.numberOnly(inp)}></input> 
                        </ToggleButton>
                    
                        <ToggleButton value={'1, 2'} style={buttonStyle}>
                            <input style={inputStyle} className="note" type="text" maxLength="1" onKeyPress={(inp) => this.numberOnly(inp)}></input> 
                        </ToggleButton>
                    
                        <ToggleButton value={'1, 3'} style={buttonStyle}>
                            <input style={inputStyle} className="note" type="text" maxLength="1" onKeyPress={(inp) => this.numberOnly(inp)}></input> 
                        </ToggleButton>
                    </ToggleButtonGroup>
                    <p style={pStyle}>|</p>
                    <ToggleButtonGroup type="checkbox" defaultValue={['', '']} style={toggleStyle}>
                        <ToggleButton value={'2, 0'} style={buttonStyle}>
                            <input style={inputStyle} className="note" type="text" maxLength="1" onKeyPress={(inp) => this.numberOnly(inp)}></input> 
                        </ToggleButton>
                    
                        <ToggleButton value={'2, 1'} style={buttonStyle}>
                            <input style={inputStyle} className="note" type="text" maxLength="1" onKeyPress={(inp) => this.numberOnly(inp)}></input> 
                        </ToggleButton>
                    
                        <ToggleButton value={'2, 2'} style={buttonStyle}>
                            <input style={inputStyle} className="note" type="text" maxLength="1" onKeyPress={(inp) => this.numberOnly(inp)}></input> 
                        </ToggleButton>
                    
                        <ToggleButton value={'2, 3'} style={buttonStyle}>
                            <input style={inputStyle} className="note" type="text" maxLength="1" onKeyPress={(inp) => this.numberOnly(inp)}></input> 
                        </ToggleButton>
                    </ToggleButtonGroup>
                    <p style={pStyle}>|</p>
                    <ToggleButtonGroup type="checkbox" defaultValue={['', '']} style={toggleStyle}>
                        <ToggleButton value={'3, 0'} style={buttonStyle}>
                            <input style={inputStyle} className="note" type="text" maxLength="1" onKeyPress={(inp) => this.numberOnly(inp)}></input> 
                        </ToggleButton>
                    
                        <ToggleButton value={'3, 1'} style={buttonStyle}>
                            <input style={inputStyle} className="note" type="text" maxLength="1" onKeyPress={(inp) => this.numberOnly(inp)}></input> 
                        </ToggleButton>
                    
                        <ToggleButton value={'3, 2'} style={buttonStyle}>
                            <input style={inputStyle} className="note" type="text" maxLength="1" onKeyPress={(inp) => this.numberOnly(inp)}></input> 
                        </ToggleButton>
                    
                        <ToggleButton value={'3, 3'} style={buttonStyle}>
                            <input style={inputStyle} className="note" type="text" maxLength="1" onKeyPress={(inp) => this.numberOnly(inp)}></input> 
                        </ToggleButton>
                    </ToggleButtonGroup>
                    <p style={pStyle}>|</p>
                </div>
                  
            </div>
        )
    }
}

export default withRouter(ScoreInput);
