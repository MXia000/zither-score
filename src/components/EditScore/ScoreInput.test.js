import React from 'react';
import { shallow } from 'enzyme';
import ScoreInput from './ScoreInput';

describe("ScoreInput", () => {
    it('render scoreinput without crashing', () => {  
        const ScoreInputComp = shallow(<div><ScoreInput /></div>);
        expect(ScoreInputComp).toMatchSnapshot();
    });
});