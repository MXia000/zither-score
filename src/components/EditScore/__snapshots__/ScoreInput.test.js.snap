// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ScoreInput render scoreinput without crashing 1`] = `ShallowWrapper {}`;
